doc/src/modules/core.rst

```diff
@@ -46,9 +46,28 @@ singleton
 
 S
 ^
-.. autoclass:: S
+.. autoclass:: Singleton
    :members:
 
+.. note:: What's the difference between `x == 1` and `x is S.One`
+
+   * S.Zero denotes a singleton object in SymPy.
```

Comment:
As per the "note", start with only discussion about S.One. In a final bullet say that the same applies to S.NegativeOne and S.Zero. As it is now, the discussion impresses me as being too fragmented.

Also watch for punctuation (spaces after periods and commas), too.